nautobot/dcim/models/devices.py

```diff
@@ -96,7 +97,7 @@ class DeviceType(PrimaryModel):
 
     manufacturer = models.ForeignKey(to="dcim.Manufacturer", on_delete=models.PROTECT, related_name="device_types")
     model = models.CharField(max_length=100)
-    slug = models.SlugField(max_length=100)
+    slug = AutoSlugField(populate_from="model", max_length=100, editable=True, overwrite_on_add=False)
```

Comment:
Ew, we don't require DeviceType.slug to be globally unique? That makes me sad. Good of you to preserve the existing behavior here though.

Reply:
Ugh.

Reply:
It uses the Meta unique_together, which caused me a ton of pain on the API front.